File: src/Core/Dating.Application.Contracts/Commands/CreateAccountCommand.cs
namespace Dating.Application.Contracts.Commands;

public class CreateAccountCommand : RequestBase
{
    public string? Email { get; set; }
    public string? Password { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? LivingCity { get; set; }
    public Gender Gender { get; set; }
    public SexualOrientation Orientation { get; set; }
    public DateTime Birthdate { get; set; }
    public string? MainPhotoUrl { get; set; }
}

File: src/Client/Dating.Web/src/app/shared/models/commands/createAccountCommand.ts
import {Gender, SexualOrientation} from '../../types';

export interface CreateAccountCommand {
  email: string;
  password: string;
  firstName: string;
  lastName: string;
  birthdate: string;
  livingCity: string;
  gender: Gender;
  orientation: SexualOrientation;
  mainPhotoUrl: string;
}
